# IMPORTATION DE LA BIBLIOTHEQUE

In [177]:
import pandas as pd


# 1-IMPORTATION DU DATASET

In [156]:
df = pd.read_csv("YugiOh - Feuille 1.csv", sep=",")
df

Nom      Type_1                  Type_2  \
0      Temple Chronomal - Trilithe       Magie                     NaN   
1               1er Mouvement Solo       Magie                     NaN   
2                  À Forces Égales       Piège                     NaN   
3                A Jamais Destinés       Piège                     NaN   
4              À la Recherche d'Or  Compétence        [ Compétence / ]   
...                            ...         ...                     ...   
10831          ZW - Lance Unicorne     Monstre        [ Bête / Effet ]   
10832     ZW - Pégase Double Sabre     Monstre        [ Bête / Effet ]   
10833      ZW - Porteur de Tornade     Monstre      [ Dragon / Effet ]   
10834          ZW - Serres d'Aigle     Monstre  [ Bête-Ailée / Effet ]   
10835           ZW - Vent Sylphide     Monstre        [ Bête / Effet ]   

         Icône                                        Description  Niveau  \
0      Continu  Si vous ne contrôlez aucun monstre, ou si tous...     NaN   
1          NaN  Si vous ne contrôlez aucun monstre : Invoquez ...     NaN   
2          NaN  À la fin de la Battle Phase, si votre adversai...     NaN   
3          NaN  Ciblez 2 monstres dans votre Cimetière ; Invoq...     NaN   
4          NaN  Activation de Compétence\n Retournez cette car...     NaN   
...        ...                                                ...     ...   
10831      NaN  Vous pouvez cibler 1 "Numéro C39 : Rayon Utopi...     4.0   
10832      NaN  Vous ne pouvez contrôler qu'1 "ZW - Pégase Dou...     5.0   
10833      NaN  Vous pouvez cibler 1 monstre ''Utopie'' que vo...     5.0   
10834      NaN  Si votre adversaire a min. 2000 Life Points de...     5.0   
10835      NaN  Vous ne pouvez contrôler qu'1 "ZW - Vent Sylph...     4.0   

      Attribut   ATK   DEF  Prix  
0          NaN   NaN   NaN  0,25  
1          NaN   NaN   NaN     6  
2          NaN   NaN   NaN     0  
3          NaN   NaN   NaN  0,25  
4          NaN   NaN   NaN   0,5  
...        ...   ...   ...   ...  
10831  Lumière  1900     0   0,3  
10832  Lumière  1000  2100  0,25  
10833     Vent  1300  1800  0,25  
10834     Vent  2000  1200  0,25  
10835  Lumière   800  1600   0,5  

[10836 rows x 10 columns]

# 2- DESCRIPTOIN DU DATASET
# Il s'agit d'un dataset qui prend son origine sur les dessins annimés comportant 10836 Lignes et 10 collonnes et comportatnt aussi des valeurs manquantes 

# 3- AFFICHAGE DES 10 PREMIERES LIGNES DU DATASET

In [157]:
df.head(10)

Nom      Type_1             Type_2       Icône  \
0  Temple Chronomal - Trilithe       Magie                NaN     Continu   
1           1er Mouvement Solo       Magie                NaN         NaN   
2              À Forces Égales       Piège                NaN         NaN   
3            A Jamais Destinés       Piège                NaN         NaN   
4          À la Recherche d'Or  Compétence   [ Compétence / ]         NaN   
5               À Moitié Fermé       Magie                NaN  Jeu Rapide   
6              A.I. Vous défie       Piège                NaN     Continu   
7          A.I. Vous Rencontre       Magie                NaN         NaN   
8                        A.I.Q       Piège                NaN     Continu   
9                        Abaki     Monstre  [ Démon / Effet ]         NaN   

                                         Description  Niveau Attribut   ATK  \
0  Si vous ne contrôlez aucun monstre, ou si tous...     NaN      NaN   NaN   
1  Si vous ne contrôlez aucun monstre : Invoquez ...     NaN      NaN   NaN   
2  À la fin de la Battle Phase, si votre adversai...     NaN      NaN   NaN   
3  Ciblez 2 monstres dans votre Cimetière ; Invoq...     NaN      NaN   NaN   
4  Activation de Compétence\n Retournez cette car...     NaN      NaN   NaN   
5  Ciblez 1 monstre face recto sur le Terrain ;il...     NaN      NaN   NaN   
6  Les monstres contrôlés par votre adversaire pe...     NaN      NaN   NaN   
7  Révélez 1 monstre Cyberse avec 2300 ATK dans v...     NaN      NaN   NaN   
8  Activez uniquement si vous contrôlez un monstr...     NaN      NaN   NaN   
9  Lorsque cette carte est détruite au combat et ...     4.0      Feu  1700   

    DEF  Prix  
0   NaN  0,25  
1   NaN     6  
2   NaN     0  
3   NaN  0,25  
4   NaN   0,5  
5   NaN   0,2  
6   NaN  0,25  
7   NaN  0,25  
8   NaN  0,25  
9  1000     0

# 4- AFFICHAGE DES 10 DERNIERES LIGNES DU DATASET

In [158]:
df.tail(10)

Nom   Type_1                    Type_2 Icône  \
10826        ZW - Armure Sleipnir  Monstre          [ Bête / Effet ]   NaN   
10827        ZW - Bouclier Ultime  Monstre        [ Rocher / Effet ]   NaN   
10828           ZW - Assaut Asura  Monstre          [ Elfe / Effet ]   NaN   
10829       ZW - Lame Foudroyante  Monstre          [ Bête / Effet ]   NaN   
10830  ZW - Hallebarde Dragonique  Monstre  [ Dragon / Effet / Xyz ]   NaN   
10831         ZW - Lance Unicorne  Monstre          [ Bête / Effet ]   NaN   
10832    ZW - Pégase Double Sabre  Monstre          [ Bête / Effet ]   NaN   
10833     ZW - Porteur de Tornade  Monstre        [ Dragon / Effet ]   NaN   
10834         ZW - Serres d'Aigle  Monstre    [ Bête-Ailée / Effet ]   NaN   
10835          ZW - Vent Sylphide  Monstre          [ Bête / Effet ]   NaN   

                                             Description  Niveau  Attribut  \
10826  Vous pouvez cibler 1 monstre "Utopie" que vous...     4.0   Lumière   
10827  Lorsque cette carte est Invoquée Normalement o...     4.0     Terre   
10828  Vous pouvez cibler 1 monstre "Utopie" que vous...     4.0       Feu   
10829  Vous pouvez cibler 1 monstre ''Utopie'' que vo...     5.0   Lumière   
10830  2 monstres de Niveau 5\n Ne peut pas attaquer ...     5.0  Ténèbres   
10831  Vous pouvez cibler 1 "Numéro C39 : Rayon Utopi...     4.0   Lumière   
10832  Vous ne pouvez contrôler qu'1 "ZW - Pégase Dou...     5.0   Lumière   
10833  Vous pouvez cibler 1 monstre ''Utopie'' que vo...     5.0      Vent   
10834  Si votre adversaire a min. 2000 Life Points de...     5.0      Vent   
10835  Vous ne pouvez contrôler qu'1 "ZW - Vent Sylph...     4.0   Lumière   

        ATK   DEF  Prix  
10826  1000  1000     0  
10827     0  2000  0,25  
10828  1000  1000   0,8  
10829  1200  2000     0  
10830  3000  1200  0,25  
10831  1900     0   0,3  
10832  1000  2100  0,25  
10833  1300  1800  0,25  
10834  2000  1200  0,25  
10835   800  1600   0,5

# 5- AFFICHAGE DES COLLONNES DU DATASET

In [160]:
df.columns

Index(['Nom', 'Type_1', 'Type_2', 'Icône', 'Description', 'Niveau', 'Attribut',
       'ATK', 'DEF', 'Prix'],
      dtype='object')

# 6- AFFICHAGE DES COLLONNES DU DATASET EN MAJISCULE

In [161]:
df.columns.str.upper()

Index(['NOM', 'TYPE_1', 'TYPE_2', 'ICÔNE', 'DESCRIPTION', 'NIVEAU', 'ATTRIBUT',
       'ATK', 'DEF', 'PRIX'],
      dtype='object')

# 7- AFFICHAGE DE LA TAILLE DU DATASET 

In [162]:
df.shape

(10836, 10)

# 8- AFFICHAGE DES INFORMATIONS SUR LES TYPES DES COLLONNES DU DATASET 

In [163]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10836 entries, 0 to 10835
Data columns (total 10 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Nom          10836 non-null  object 
 1   Type_1       10836 non-null  object 
 2   Type_2       7123 non-null   object 
 3   Icône        1863 non-null   object 
 4   Description  10735 non-null  object 
 5   Niveau       6903 non-null   float64
 6   Attribut     7021 non-null   object 
 7   ATK          6959 non-null   object 
 8   DEF          6959 non-null   object 
 9   Prix         10836 non-null  object 
dtypes: float64(1), object(9)
memory usage: 846.7+ KB


# 8.1- VERIFICATION DES VALEURS MANQUANTES DANS LE DATASET 

In [164]:
df.isnull().sum()

Nom               0
Type_1            0
Type_2         3713
Icône          8973
Description     101
Niveau         3933
Attribut       3815
ATK            3877
DEF            3877
Prix              0
dtype: int64

# 8.2 OUI IL YA LES VALEURS MANQUANTES DANS LES COLLONNES SUIVANTES:
# Type_2
# Icône
# escription
# Niveau
# Attribut
# ATK
# DEF


# 9- AFFICHAGE DES OCCURENCES DE LA COLONNE 'Type_2'

In [165]:
df["Type_2"].value_counts()

[ Guerrier / Effet ]            543
[ Machine / Effet ]             451
[ Démon / Effet ]               365
[ Magicien / Effet ]            332
[ Elfe / Effet ]                280
                               ... 
[ Aqua / Spirit / Effet ]         1
[ Insecte / Fusion / Effet ]      1
[ Effet / Plante ]                1
[ Reptile / Syntoniseur ]         1
[ Dragon / Effet / Xyz ]          1
Name: Type_2, Length: 468, dtype: int64

# 10- AFFICHAGE DES 5 PREMIERES LIGNES DE LA COLONNE 'Description'

In [166]:
df1=df["Description"]
df1
df1.head(5)

0    Si vous ne contrôlez aucun monstre, ou si tous...
1    Si vous ne contrôlez aucun monstre : Invoquez ...
2    À la fin de la Battle Phase, si votre adversai...
3    Ciblez 2 monstres dans votre Cimetière ; Invoq...
4    Activation de Compétence\n Retournez cette car...
Name: Description, dtype: object

# 11- AFFICHAGE DES 5 PREMIERES LIGNES DES COLONNES 'Description','Type_2' et 'Type_1'

In [167]:
A=df["Description"]
B=df["Type_1"]
C=df["Type_2"]
df2= pd.concat([A, B, C], axis=1)
df2.head(5)

Description      Type_1  \
0  Si vous ne contrôlez aucun monstre, ou si tous...       Magie   
1  Si vous ne contrôlez aucun monstre : Invoquez ...       Magie   
2  À la fin de la Battle Phase, si votre adversai...       Piège   
3  Ciblez 2 monstres dans votre Cimetière ; Invoq...       Piège   
4  Activation de Compétence\n Retournez cette car...  Compétence   

             Type_2  
0               NaN  
1               NaN  
2               NaN  
3               NaN  
4  [ Compétence / ]

# 12- AFFICHAGE DU CONTENU DE LA PREMIERE LIGNE ET DE LA CINQUIEME COLONNE

In [168]:
D=df["Description"]
D[0]

'Si vous ne contrôlez aucun monstre, ou si tous les monstres que vous contrôlez sont des monstres "Chronomal" : vous pouvez payer 500 LP ; immédiatement après la résolution de cet effet, Invoquez Normalement 1 monstre "Chronomal" depuis votre main. Si un Monstre Xyz "Chronomal" ou "Numéro" que vous contrôlez va activer un effet en détachant du Matériel, vous pouvez envoyer cette carte au Cimetière au lieu d\'1 de ces Matériels. Vous ne pouvez utiliser chaque effet de "Temple Chronomal - Trilithe" qu\'une fois par tour.'

In [169]:
df.iloc[0,4]

'Si vous ne contrôlez aucun monstre, ou si tous les monstres que vous contrôlez sont des monstres "Chronomal" : vous pouvez payer 500 LP ; immédiatement après la résolution de cet effet, Invoquez Normalement 1 monstre "Chronomal" depuis votre main. Si un Monstre Xyz "Chronomal" ou "Numéro" que vous contrôlez va activer un effet en détachant du Matériel, vous pouvez envoyer cette carte au Cimetière au lieu d\'1 de ces Matériels. Vous ne pouvez utiliser chaque effet de "Temple Chronomal - Trilithe" qu\'une fois par tour.'

# 13- AFFICHAGE DU CONTENU DES 5 PREMIERES LIGNES DES COLONNES 5 A 7 

In [170]:
E=df["Description"]
F=df["Niveau"]
G=df["Attribut"]
df3= pd.concat([E, F, G], axis=1)
df3.head(5)

Description  Niveau Attribut
0  Si vous ne contrôlez aucun monstre, ou si tous...     NaN      NaN
1  Si vous ne contrôlez aucun monstre : Invoquez ...     NaN      NaN
2  À la fin de la Battle Phase, si votre adversai...     NaN      NaN
3  Ciblez 2 monstres dans votre Cimetière ; Invoq...     NaN      NaN
4  Activation de Compétence\n Retournez cette car...     NaN      NaN

# 14- COPIE DU  DATASET D'ORIGINE DANS UNE VARIABLE AVEC LES VALEURS DE LA COLONNE Type_1= monstre

In [171]:
df['Type_1']= 'monstre'
df

Nom   Type_1                  Type_2    Icône  \
0      Temple Chronomal - Trilithe  monstre                     NaN  Continu   
1               1er Mouvement Solo  monstre                     NaN      NaN   
2                  À Forces Égales  monstre                     NaN      NaN   
3                A Jamais Destinés  monstre                     NaN      NaN   
4              À la Recherche d'Or  monstre        [ Compétence / ]      NaN   
...                            ...      ...                     ...      ...   
10831          ZW - Lance Unicorne  monstre        [ Bête / Effet ]      NaN   
10832     ZW - Pégase Double Sabre  monstre        [ Bête / Effet ]      NaN   
10833      ZW - Porteur de Tornade  monstre      [ Dragon / Effet ]      NaN   
10834          ZW - Serres d'Aigle  monstre  [ Bête-Ailée / Effet ]      NaN   
10835           ZW - Vent Sylphide  monstre        [ Bête / Effet ]      NaN   

                                             Description  Niveau Attribut  \
0      Si vous ne contrôlez aucun monstre, ou si tous...     NaN      NaN   
1      Si vous ne contrôlez aucun monstre : Invoquez ...     NaN      NaN   
2      À la fin de la Battle Phase, si votre adversai...     NaN      NaN   
3      Ciblez 2 monstres dans votre Cimetière ; Invoq...     NaN      NaN   
4      Activation de Compétence\n Retournez cette car...     NaN      NaN   
...                                                  ...     ...      ...   
10831  Vous pouvez cibler 1 "Numéro C39 : Rayon Utopi...     4.0  Lumière   
10832  Vous ne pouvez contrôler qu'1 "ZW - Pégase Dou...     5.0  Lumière   
10833  Vous pouvez cibler 1 monstre ''Utopie'' que vo...     5.0     Vent   
10834  Si votre adversaire a min. 2000 Life Points de...     5.0     Vent   
10835  Vous ne pouvez contrôler qu'1 "ZW - Vent Sylph...     4.0  Lumière   

        ATK   DEF  Prix  
0       NaN   NaN  0,25  
1       NaN   NaN     6  
2       NaN   NaN     0  
3       NaN   NaN  0,25  
4       NaN   NaN   0,5  
...     ...   ...   ...  
10831  1900     0   0,3  
10832  1000  2100  0,25  
10833  1300  1800  0,25  
10834  2000  1200  0,25  
10835   800  1600   0,5  

[10836 rows x 10 columns]

# 15- AFFICHGE DES INFORMATIONS SUR CETTE VARIABLE

In [172]:
df['Type_1'].info

<bound method Series.info of 0        monstre
1        monstre
2        monstre
3        monstre
4        monstre
          ...   
10831    monstre
10832    monstre
10833    monstre
10834    monstre
10835    monstre
Name: Type_1, Length: 10836, dtype: object>

# 16- AFFICHAGE DES VALEURS DU DATASET AVEC LES VALEURS DE Type_1= magie et icone =  continu

In [173]:
df.rename(columns = {'Type_1': 'magie', 'Icône':'continu'},inplace=True)
df

Nom    magie                  Type_2  continu  \
0      Temple Chronomal - Trilithe  monstre                     NaN  Continu   
1               1er Mouvement Solo  monstre                     NaN      NaN   
2                  À Forces Égales  monstre                     NaN      NaN   
3                A Jamais Destinés  monstre                     NaN      NaN   
4              À la Recherche d'Or  monstre        [ Compétence / ]      NaN   
...                            ...      ...                     ...      ...   
10831          ZW - Lance Unicorne  monstre        [ Bête / Effet ]      NaN   
10832     ZW - Pégase Double Sabre  monstre        [ Bête / Effet ]      NaN   
10833      ZW - Porteur de Tornade  monstre      [ Dragon / Effet ]      NaN   
10834          ZW - Serres d'Aigle  monstre  [ Bête-Ailée / Effet ]      NaN   
10835           ZW - Vent Sylphide  monstre        [ Bête / Effet ]      NaN   

                                             Description  Niveau Attribut  \
0      Si vous ne contrôlez aucun monstre, ou si tous...     NaN      NaN   
1      Si vous ne contrôlez aucun monstre : Invoquez ...     NaN      NaN   
2      À la fin de la Battle Phase, si votre adversai...     NaN      NaN   
3      Ciblez 2 monstres dans votre Cimetière ; Invoq...     NaN      NaN   
4      Activation de Compétence\n Retournez cette car...     NaN      NaN   
...                                                  ...     ...      ...   
10831  Vous pouvez cibler 1 "Numéro C39 : Rayon Utopi...     4.0  Lumière   
10832  Vous ne pouvez contrôler qu'1 "ZW - Pégase Dou...     5.0  Lumière   
10833  Vous pouvez cibler 1 monstre ''Utopie'' que vo...     5.0     Vent   
10834  Si votre adversaire a min. 2000 Life Points de...     5.0     Vent   
10835  Vous ne pouvez contrôler qu'1 "ZW - Vent Sylph...     4.0  Lumière   

        ATK   DEF  Prix  
0       NaN   NaN  0,25  
1       NaN   NaN     6  
2       NaN   NaN     0  
3       NaN   NaN  0,25  
4       NaN   NaN   0,5  
...     ...   ...   ...  
10831  1900     0   0,3  
10832  1000  2100  0,25  
10833  1300  1800  0,25  
10834  2000  1200  0,25  
10835   800  1600   0,5  

[10836 rows x 10 columns]

# 17- CONVERTION DES COLONNES ATK ET DEF EN ENTIERT ET LA COLONNE PRIX EN FLOAT 

# 17- Pour convertir, il faut tout d'abord supprimer les valeurs manquantes dans ces colonnes

In [ ]:
df["ATK"]=df["ATK"].astype(int)
df["DEF"]=df["DEF"].astype(int)
df["Prix"]=df["Prix"].astype(float)

# 18- SUPPRESSION DES VALEURS MANQUANTES DANS LES COLONNES ATK ET Type_2

In [175]:
H=df["Type_2"]
I=df["ATK"]
df6= pd.concat([H,I], axis=1)
df6.dropna()

Type_2   ATK
9                   [ Démon / Effet ]  1700
10          [ Bête-Guerrier / Effet ]  1200
11     [ Bête / Syntoniseur / Effet ]  1000
12                [ Insecte / Effet ]   100
13                [ Insecte / Effet ]  1600
...                               ...   ...
10831                [ Bête / Effet ]  1900
10832                [ Bête / Effet ]  1000
10833              [ Dragon / Effet ]  1300
10834          [ Bête-Ailée / Effet ]  2000
10835                [ Bête / Effet ]   800

[6959 rows x 2 columns]